frontend/src/app/commonComponents/SingleFileInput.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { ChangeEventHandler, useState } from "react";
+import classnames from "classnames";
+
+import "./SingleFileInput.scss";
+
+type SingleFileInputProps = {
+  id: string;
+  name: string;
+  ariaLabel: string;
+  accept?: string;
+  onChange: ChangeEventHandler<HTMLInputElement>;
+  required: boolean;
+};
+
+type FileState = "blank" | "invalid" | "selected";
+
+const SingleFileInput = ({
+  id,
+  name,
+  onChange,
+  ariaLabel,
+  accept,
+  required = false,
+}: SingleFileInputProps) => {
+  const [fileState, setFileState] = useState<FileState>("blank");
+
+  function handleOnChange(e: React.ChangeEvent<HTMLInputElement>) {
+    if (e?.currentTarget?.files?.length === 0) {
+      setFileState("blank");
+    } else {
+      const file = e?.currentTarget?.files?.item(0);
+
+      if (file?.type.search(/csv/i) === -1) {
+        setFileState("invalid");
+      } else {
+        setFileState("selected");
+      }
+    }
+
+    onChange(e);
+  }
+
+  function getHint(fileState: FileState): string {
+    switch (fileState) {
+      case "invalid":
+        return "This is not a valid file type";
+      case "selected":
+        return "Drag files here or choose from folder to change file";
```

Comment:
I think this should be just "Drag file here or choose from folder to change file"  (file singular) since we only support one file.

Reply:
Totally on point, thanks for the suggestion :)